src/main/java/com/ltonetwork/client/core/transacton/SetScript.java

```diff
@@ -17,6 +18,12 @@ public SetScript(String script) {
         this.script = (script == null) ? null : script.replaceAll("^(base64:)?", "base64:");
     }
 
+    public SetScript(JsonObject json) {
```

Comment:
The JsonObject can either be the full tx json or result of `compile`. Either work it out here, or add another function.

Reply:
I've added `complexity` and `extraFee` attributes to `SetScript`. If I am not mistaken, those are the only new things we receive from running `compile` and none of the rest is skipped/different.